WordPress/Classes/Stores/StatsPeriodStore.swift

```diff
@@ -1402,12 +1402,22 @@ private extension PeriodStoreState {
                                                                  url: url,
                                                                  timeZone: timeZone,
                                                                  date: Date(timeIntervalSinceReferenceDate: 0),
-                                                                 stats: ThisWeekWidgetStats())
+                                                                 stats: ThisWeekWidgetStats(days: initializedWeekdays))
 
             }
         }
     }
 
+    // creates a list of days from the current date with empty stats to avoid showing an empty widget preview
+    var initializedWeekdays: [ThisWeekWidgetDay] {
+        let secondsPerDay = 86400.0
+        var days = [ThisWeekWidgetDay]()
+        for index in 0...7 {
+            days.insert(ThisWeekWidgetDay(date: Date(timeIntervalSinceNow: -(Double(index) * secondsPerDay)), viewsCount: 0, dailyChangePercent: 0), at: index)
```

Comment:
I wonder if it might be best to use something like `NSCalendar.current.date(byAdding: .day, value: -index, to: Date())` rather than hardcoding seconds per day?

Reply:
Done